spec/mysql2/client_spec.rb

```diff
@@ -172,7 +172,16 @@ def run_gc
 
   it "should terminate connections when calling close" do
     expect {
-      Mysql2::Client.new(DatabaseCredentials['root']).close
+      client = Mysql2::Client.new(DatabaseCredentials['root'])
+      connection_id = client.thread_id
+      client.close
+
+      # mysql_close sends a quit command without waiting for a response
+      # so give the server some time to handle the detect the closed connection
+      10.times do
+        break unless @client.query("SHOW PROCESSLIST").detect { |row| row['Id'] == connection_id }
+        sleep(0.1)
+      end
     }.to_not change {
       @client.query("SHOW STATUS LIKE 'Aborted_%'").to_a +
         @client.query("SHOW STATUS LIKE 'Threads_connected'").to_a
```

Comment:
`Threads_connected` *should* change, huh?

Perhaps we should just check that the client connection wasn't aborted. The `SHOW PROCESSLIST` check is already demonstrating that the client is no longer connected, which is the aim of the spec.